charms/jupyter-controller/tests/integration/test_charm.py

```diff
@@ -161,3 +164,38 @@ async def test_prometheus_integration(ops_test: OpsTest):
     wait=tenacity.wait_exponential(multiplier=1, min=1, max=10),
     reraise=True,
 )
+
+
+@pytest.mark.abort_on_fail
+async def test_remove_with_resources_present(ops_test: OpsTest):
```

Comment:
I like this test.  I wonder if we can extend it to also make sure user's notebooks are cleaned up when the controller is removed